packages/react-router/lib/hooks.tsx

```diff
@@ -935,8 +936,9 @@ let blockerId = 0;
 export function useBlocker(shouldBlock: boolean | BlockerFunction): Blocker {
   let { router } = useDataRouterContext(DataRouterHook.UseBlocker);
   let state = useDataRouterState(DataRouterStateHook.UseBlocker);
-  let [blockerKey] = React.useState(() => String(++blockerId));
```

Comment:
Moved this increment into a side effect to be compliant with `StrictMode`.  Fetchers probably have a similar issue but they don't need to put anything like a blocker function into an internal router cache so it doesn't end up causing any user-facing issues.